File: 01_Language/01_Functions/python/preg_grep.py
# coding: utf-8

import re


def preg_grep(pattern, array):
    p = re.compile(pattern)
    return filter(lambda x: p.search(x), array)


if __name__ == '__main__':
    arr = ['1.23', '.45', 'abc']
    print(list(preg_grep(r'^(\d+)?\.\d+$', arr)))
    print(list(preg_grep(r'\d+', arr)))

File: 01_Language/01_Functions/java/src/main/java/com/onevgo/functions/PregGrep.java
package com.onevgo.functions;

import java.util.Arrays;
import java.util.Collection;
import java.util.List;
import java.util.regex.Pattern;
import java.util.stream.Collectors;

public class PregGrep {
    public static List<String> pregGrep(String pattern, Collection<String> array) {
        Pattern compile = Pattern.compile(pattern);
        return array.stream().filter(s -> compile.matcher(s).find()).collect(Collectors.toList());
    }

    public static void main(String[] args) {
        List<String> array = Arrays.asList("1.23", ".45", "abc");
        System.out.println(pregGrep("^(\\d+)?\\.\\d+$", array));
        System.out.println(pregGrep("\\d+", array));
    }
}
